### PR TITLE
Add changelog for v3.1.0

### PR DESCRIPTION
**Pull request by trevorflahardy**
PR adds the changelog for the upcoming `v3.1.0` update to the library.

Conventionally, it is the PR's responsibility to update the changelog; but, because this was accidentally missed, this PR has been created as a one-time exception for updating the documentation for any new changes. 

PR status as of October 19th, 2024:
- [x] Waiting on #28 to be merged beforehand.

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -1,5 +1,29 @@
...
+-------
+
+New Features
+~~~~~~~~~~~~~
+- Add :attr:`fortnite_api.MaterialInstance.product_tag`
+- Add :attr:`fortnite_api.ShopEntry.colors`
+- Add :attr:`fortnite_api.ShopEntry.offer_tag`
+- Add :attr:`fortnite_api.ShopEntryLayout.rank`
+- Add :attr:`fortnite_api.ShopEntryNewDisplayAsset.render_images`
+- Create new object :class:`fortnite_api.ShopEntryRenderImage`
```
Comment (trevorflahardy) [inline docs/changelog.rst]: This conflicts with #26. I recommend removing these lines here, merging this PR, and leaving it up to #26 to update this changelog with its changes.
```suggestion
```

**Comment by trevorflahardy** [comment]
Note that the `docs` test will fail until #28 has been merged into the master branch because this PR references the `response_flags` field in the documentation. Ensure docs pass after #28 has been merged before merging this PR.

File: docs/changelog.rst [hunk 2]
```diff
@@ -1,5 +1,52 @@
...
+.. _vp3p1p0:
+
+v3.1.0
+-------
+This version introduces new data for shop-related objects, reflecting the updated shop layouts and the Fortnite webshop. Additionally, it includes functions that were omitted in version 3.0.0 and addresses a design decision that results in a breaking change.
+
+Breaking Changes
+~~~~~~~~~~~~~~~~
+- ``ShopEntryNewDisplayAsset`` has been renamed to :class:`fortnite_api.NewDisplayAsset`.
+- ``BannerColor.colour`` has been removed as it was merely an alias for :attr:`fortnite_api.BannerColor.color`.
```
Comment (trevorflahardy) [inline docs/changelog.rst]:
```suggestion
- Alias ``BannerColor.colour`` has been removed for consistency. The API does not use aliases, use  attr:`fortnite_api.BannerColor.color` instead.
```

File: docs/changelog.rst [hunk 3]
```diff
@@ -1,5 +1,52 @@
...
 Changelog
 =========
-This page keeps track of all the changes to the project. For help on Migrating to Version 3 from Version 2, see the :ref:`Migration guide <migrating>`.
\ No newline at end of file
+
+.. _vp3p1p0:
+
+v3.1.0
+-------
+This version introduces new data for shop-related objects, reflecting the updated shop layouts and the Fortnite webshop. Additionally, it includes functions that were omitted in version 3.0.0 and addresses a design decision that results in a breaking change.
```
Comment (trevorflahardy) [inline docs/changelog.rst]:
```suggestion
This version introduces new data for shop-related objects, reflecting the updated shop layouts and the Fortnite webshop. Additionally, it includes functions that were omitted in version v3.0.0 and addresses a design decision that results in a breaking change.
```

File: docs/changelog.rst [hunk 4]
```diff
@@ -1,5 +1,52 @@
...
+- Added attribute :attr:`fortnite_api.ShopEntryLayout.rank`.
+- Added attribute :attr:`fortnite_api.NewDisplayAsset.render_images`.
+- Added attribute :attr:`fortnite_api.ShopEntry.offer_tag`.
+- Added attribute :attr:`fortnite_api.ShopEntry.colors`.
+
+Bug Fixes
+~~~~~~~~~
+- Fixed an issue where ``type`` and ``time_window`` parameters were not respected when fetching stats.
+- :attr:`fortnite_api.Playlist.images` now returns ``None`` when no images are available, instead of an empty dict.
+- Ensured all datetime objects include timezone information to avoid returning naive datetime objects in rare cases.
```
Comment (trevorflahardy) [inline docs/changelog.rst]:
```suggestion
- Bug fix for returning naive datetime objects in rare cases. All datetime objects are UTC aware.
```

File: docs/changelog.rst [hunk 5]
```diff
@@ -1,5 +1,52 @@
...
+
+Documentation
+~~~~~~~~~~~~~
+- Added :ref:`response flags <response_flags>` documentation to explain how to use the ``fortnite_api.ResponseFlags`` class, how to enable response flags, which response flags are available, and when you should enable them.
+- Added ``opt-in`` directive on attributes that require a specific response flag to be set. This is to ensure users are aware of the response flags required to access certain attributes when using the API.
```
Comment (trevorflahardy) [inline docs/changelog.rst]:
```suggestion
- Added ``opt-in`` directive in the documentation on attributes that require a specific response flag to be set. This ensures users know of the response flags required to access certain attributes when using the API.
```

File: docs/changelog.rst [hunk 6]
```diff
@@ -1,5 +1,52 @@
...
+
+Miscellaneous
+~~~~~~~~~~~~~
+- Fetching specific game mode news raised :class:`fortnite_api.ServiceUnavailable`, due to improper handling from Fortnite-API.com. This has been fixed within the API. Now, when no news is available, :class:`fortnite_api.NotFound` is raised instead. This change is reflected in the documentation.
```
Comment (trevorflahardy) [inline docs/changelog.rst]:
```suggestion
- Previously, fetching specific game mode news raised :class:`fortnite_api.ServiceUnavailable` due to improper handling from Fortnite-API.com. This has been fixed within the API. Now, when no news is available, :class:`fortnite_api.NotFound` is raised instead. This change is also reflected in the documentation.
```

File: docs/changelog.rst [hunk 7]
```diff
@@ -1,5 +1,52 @@
...
+- Improved documentation for attributes that require specific response flags to be set.
```
Comment (trevorflahardy) [inline docs/changelog.rst]: Redundant from above
```suggestion
```

**Review by trevorflahardy — COMMENTED** [review]
Just some thoughts I saw while reviewing - instead of directly committing and merging I wanted to run them by you as well. Thanks 👍

**Review by Luc1412 — APPROVED** [review]

**Pull request merged**